### PR TITLE
[BUGFIX] Hashing with argon2i as default

### PR DESCRIPTION
**Pull request by Footrotflat**

### CODE REVIEW

**Comment by TheNaderio** [comment]
It is not argon2i as default but the configured hashing thats set in the installTool - isn't it?

**Comment by norderik** [comment]
In V9 argon2i is default in the installTool. 
Since femanager 5.0 is not compatible with 8.7 the code referring to saltedpasswords and sha1 is obsolet.

**Comment by sgrossberndt** [comment]
Without this change changing the password when editing a user in TYPO3 v9 is not possible.
After setting a new password in 5.0.0 and trying to log in in felogin an `InvalidPasswordHashException` is thrown which even reveals the entered password in the debug exception: `at TYPO3\CMS\Core\Crypto\PasswordHashing\PasswordHashFactory->get('mysecretpassword', 'FE')`

**Comment by sgrossberndt** [comment]
@sbusemann This pull request should be included in 5.0.1 as well

File: Classes/Utility/UserUtility.php [hunk 1]
```diff
@@ -10,8 +10,7 @@
 use TYPO3\CMS\Extbase\Configuration\ConfigurationManagerInterface;
```
Comment (sgrossberndt) [inline Classes/Utility/UserUtility.php]: The class `ExtensionManagementUtility` is now unused as well in this class, its use statement can be removed.

**Comment by sgrossberndt** [comment]
Fixes #146

File: Classes/Utility/UserUtility.php [hunk 2]
```diff
@@ -182,12 +181,8 @@ public static function hashPassword(User &$user, $method)
                 break;
 
             default:
-                if (ExtensionManagementUtility::isLoaded('saltedpasswords')) {
-                    if (SaltedPasswordsUtility::isUsageEnabled('FE')) {
-                        $objInstanceSaltedPw = SaltFactory::getSaltingInstance();
-                        $user->setPassword($objInstanceSaltedPw->getHashedPassword($user->getPassword()));
-                    }
-                }
```
Comment (KonstantinSoelch) [inline Classes/Utility/UserUtility.php]: Just cause i saw it, while i tried the PR.
The PhpDoc is missing `/** @var PasswordHashInterface $hashInstance */`

Comment (sgrossberndt) [reply Classes/Utility/UserUtility.php]: That is not necessary as `dynamicReturnType` added in https://forge.typo3.org/issues/63553 can display which class is created by makeInstance. 

`DynamicReturnTypePlugin` in PHPStorm.

**Comment by sbusemann** [comment]
In order to get version 5 of femanager also compatible for TYPO3 8.7, we switched to another solution.

**Pull request closed without merging**